lib/nat/service.go

```diff
@@ -28,24 +28,28 @@ import (
 type Service struct {
 	suture.Service
 
-	id  protocol.DeviceID
-	cfg config.Wrapper
+	id               protocol.DeviceID
+	cfg              config.Wrapper
+	processScheduled chan struct{}
 
 	mappings []*Mapping
-	timer    *time.Timer
+	enabled  bool
 	mut      sync.RWMutex
 }
 
 func NewService(id protocol.DeviceID, cfg config.Wrapper) *Service {
 	s := &Service{
-		id:  id,
-		cfg: cfg,
+		id:               id,
+		cfg:              cfg,
+		processScheduled: make(chan struct{}, 1),
 
-		timer: time.NewTimer(0),
-		mut:   sync.NewRWMutex(),
+		mut: sync.NewRWMutex(),
 	}
 	s.Service = util.AsService(s.serve, s.String())
 	cfg.Subscribe(s)
+	s.mut.Lock()
+	s.enabled = cfg.Options().NATEnabled
+	s.mut.Unlock()
```

Comment:
I'd just call s.CommitConfiguration(cfg, cfg) like we do elsewhere.

Reply:
Personally I consider that pattern bag shit ugly (in principle nothing changed, so you might even argue it should be a noop). Changed it anyway, as it's indeed how we do stuff in many places (and it works here).